crates/modelardb_storage/src/data_folder.rs

```diff
@@ -0,0 +1,1681 @@
+        DeltaTableWriter::try_new(delta_table, partition_columns, writer_properties)
+    }
+
+    /// Return a [`DeltaTableWriter`] for writing to the table with `delta_table` in the Delta Lake,
+    /// or a [`ModelarDbStorageError`] if a connection to the Delta Lake cannot be established or
+    /// the table does not exist.
+    pub async fn normal_or_metadata_table_writer(
+        &self,
+        delta_table: DeltaTable,
+    ) -> Result<DeltaTableWriter> {
+        let writer_properties = apache_parquet_writer_properties(None);
+        DeltaTableWriter::try_new(delta_table, vec![], writer_properties)
+    }
+
+    /// Create a Delta Lake table for a metadata table with `table_name` and `schema` if it does not
+    /// already exist. If the metadata table could not be created, [`ModelarDbStorageError`] is
+    /// returned. An error is not returned if the metadata table already exists.
+    pub async fn create_metadata_table(
+        &self,
+        table_name: &str,
+        schema: &Schema,
+    ) -> Result<DeltaTable> {
+        self.create_table(
+            table_name,
+            schema,
+            &[],
+            self.location_of_metadata_table(table_name),
+            SaveMode::Ignore,
+        )
+        .await
+    }
+
+    /// Return the location of the metadata table with `table_name`.
+    fn location_of_metadata_table(&self, table_name: &str) -> String {
+        format!("{}/{METADATA_FOLDER}/{table_name}", self.location)
+    }
+
+    /// Create a Delta Lake table for a normal table with `table_name` and `schema` if it does not
+    /// already exist. If the normal table could not be created, e.g., because it already exists,
+    /// [`ModelarDbStorageError`] is returned.
+    pub async fn create_normal_table(
+        &self,
+        table_name: &str,
+        schema: &Schema,
+    ) -> Result<DeltaTable> {
+        self.create_table(
+            table_name,
+            schema,
+            &[],
+            self.location_of_compressed_table(table_name),
+            SaveMode::ErrorIfExists,
+        )
+        .await
+    }
+
+    /// Create a Delta Lake table for a time series table with `time_series_table_metadata` if it
+    /// does not already exist. Returns [`DeltaTable`] if the table could be created and
+    /// [`ModelarDbStorageError`] if it could not.
+    pub async fn create_time_series_table(
+        &self,
+        time_series_table_metadata: &TimeSeriesTableMetadata,
+    ) -> Result<DeltaTable> {
+        self.create_table(
+            &time_series_table_metadata.name,
+            &time_series_table_metadata.compressed_schema,
+            &[FIELD_COLUMN.to_owned()],
+            self.location_of_compressed_table(&time_series_table_metadata.name),
+            SaveMode::ErrorIfExists,
+        )
+        .await
+    }
+
+    /// Return the location of the compressed time series or normal table with `table_name`.
+    fn location_of_compressed_table(&self, table_name: &str) -> String {
```

Comment:
Similarly to above, this could be renamed to `location_of_table`.

Reply:
Fixed.